things/service.go

```diff
@@ -530,6 +530,30 @@ func (svc service) ListClientsByGroup(ctx context.Context, token, groupID string
 	}, nil
 }
 
+func (svc service) SearchThings(ctx context.Context, token string, pm mgclients.Page) (mgclients.ClientsPage, error) {
+	res, err := svc.identify(ctx, token)
+	if err != nil {
+		return mgclients.ClientsPage{}, err
+	}
+
+	cp, err := svc.clients.SearchBasicInfo(ctx, pm)
+	if err != nil {
+		return mgclients.ClientsPage{}, err
+	}
+
+	things := mgclients.ClientsPage{}
+	for _, val := range cp.Clients {
+		if _, err := svc.authorize(ctx, res.GetDomainId(), auth.UserType, auth.UsersKind, res.GetId(), auth.ViewPermission, auth.ThingType, val.ID); err == nil {
+			things.Clients = append(things.Clients, val)
+			things.Page.Total += 1
+		}
+	}
```

Comment:
This will not return proper page items.

Reply:
I think we don't need this `SearchThings` function in service layer

Reply:
I will remove the `SearchThings` function then.

Reply:
The authorization is needed because the `member` should not be able to `view` things created by other members

Reply:
Yeah, In `ListThings`  it is handled.